app/uk/gov/hmrc/gform/validation/TextChecker.scala

```diff
@@ -207,9 +208,19 @@ object TextChecker {
 
     def lookupError: CheckProgram[Unit] = {
       val vars: List[String] = lookupLabel.label :: Nil
-      validationFailure(fieldValue, genericErrorLookup, Some(vars))
+      validationFailure(fieldValue, getLookupErrorMessageKey, Some(vars))
     }
 
+    def getLookupErrorMessageKey: String = {
+      for {
+        text   <- fieldValue.`type`.cast[Text]
+        lookup <- text.constraint.cast[Lookup]
+      } yield lookup.register match {
+        case Register.Country => countryErrorLookup
+        case _                => genericErrorLookup
+      }
+    }.getOrElse(genericErrorLookup)
+
```

Comment:
```suggestion
    def getLookupErrorMessageKey: String =
      fieldValue match {
        case IsText(Text(Lookup(Register.Country, _), _, _, _, _, _)) => countryErrorLookup
        case _                                                        => genericErrorLookup
      }
```

Reply:
Here is a suggestion with reusing custom matcher `IsText` gform provides and uses extensively. It would be nice to keep using it for consistency.

Reply:
Thanks @josef-vlach - makes perfect sense to use this existing custom matcher